### DISCUSSION TITLE
Proposal: add method to Storage/Filesystem or somewhere, to get image dimensions using the getimagesize php function

### DISCUSSION THREAD

**Discussion by terremoth** [laravel/framework#41766] [Ideas]
Example:

```php
// return getimagesize default array
Storage::disk('public')->image('flower.jpg')->dimensions();
```
or

```php
Storage::disk('public')->image('flower.jpg')->width();
Storage::disk('public')->image('flower.jpg')->height();
```
I don't know if the ```Storage``` class is the right place to have this, but in the end I just want to have a beauty expressive and clean way to get image dimensions, the Laravel Way.

**Comment by a-bashtannik** [comment]
Breaks the single responsibility principle. 
Use something like [Image Intervention](https://image.intervention.io/v2).

**Reply by terremoth** [reply]
I know it breaks.
But your proposal is to use another package...

If it was to use another package, I wouldn't be creating a discussion here.
